Tests/SentryTests/SentrySessionTests.swift

```diff
@@ -0,0 +1,34 @@
+import XCTest
+
+class SentrySessionTestsSwift: XCTestCase {
+    
+    private var currentDateProvider: TestCurrentDateProvider!
+    
+    override func setUp() {
+        currentDateProvider = TestCurrentDateProvider()
+        CurrentDate.setCurrentDateProvider(currentDateProvider)
+    }
+    
+    func testEndSession() {
+        let session = SentrySession()
+        let date = currentDateProvider.date().addingTimeInterval(1)
+        session.endExited(withTimestamp: date)
+        
+        XCTAssertEqual(1, session.duration)
+        XCTAssertEqual(date, session.timestamp)
+        XCTAssertEqual(SentrySessionStatus.exited, session.status)
+    }
+    
+    func testInitAndDurationNilWhenSerialize() {
+        let session = SentrySession()
+        session.setInit(nil)
+        
+        let date = currentDateProvider.date()
+        session.endExited(withTimestamp: date.addingTimeInterval(2))
+        session.duration = nil
+        
+        currentDateProvider.setDate(date: date.addingTimeInterval(3))
+        let sessionSerialized = session.serialize()
```

Comment:
I'd propose we remove all setters. It's fine to have them public for testing but the goal was not to let users mutate the inner state
To be honest I don't think it's ideal to have side effect when calling serialize. I just wrote this to guarantee session had duration at the time of serialization insead of exposing some hacky `calculateDuration` which the caller would have to call before calling serialize.

Reply:
That makes sense, but it would be a breaking change. As long as we don't bump a major version I would like to keep the tests here to make sure our implementation is working.

Reply:
There's no real API to access these properties anyway so we could fit this into 5.1

Reply:
Okay, I would do this in another PR, if that is fine for you.

Reply:
👍